### PR TITLE
Make Astro Optional

### PR DESCRIPTION
**Pull request by lishaduck**
_**Closes: #153**_
_**Closes: #130**_

Put Astro behind an option.

### Tests

> It works in my vanilla-TS Bun monorepo; is there a good Astro project I can test it on?

_Originally posted by @lishaduck in https://github.com/AndreaPontrandolfo/sheriff/issues/153#issuecomment-2218249341_

### CODE REVIEW

**Comment by lishaduck** [comment]
Oh, and if this does work, then I'll need to update the docs.

**Comment by AndreaPontrandolfo** [comment]
Yup, can you update the docs?

**Comment by lishaduck** [comment]
Oops.

File: packages/eslint-config-sheriff/src/getReactConfig.ts [hunk 1]
```diff
@@ -57,7 +57,7 @@ export const getReactConfig = (customTSConfigPath?: string | string[]) => {
       plugins: {
         'jsx-a11y': reactAccessibility,
       },
-      rules: reactAccessibility.configs.recommended.rules,
+      rules: reactAccessibility.configs.strict.rules,
```
Comment (AndreaPontrandolfo) [inline packages/eslint-config-sheriff/src/getReactConfig.ts:60]: why this?

File: packages/eslint-config-sheriff/src/getBaseConfig.ts [hunk 2]
```diff
@@ -138,6 +136,7 @@ export const getBaseConfig = (userConfigChoices: SheriffSettings) => {
       settings: {
         'import/parsers': {
           '@typescript-eslint/parser': ['.ts', '.tsx'],
+          espree: ['.js'],
```
Comment (AndreaPontrandolfo) [inline packages/eslint-config-sheriff/src/getBaseConfig.ts:139]: what is this for?

File: apps/docs-website/docs/vscode-support.md [hunk 3]
```diff
@@ -38,10 +38,18 @@ For [Astro](https://astro.build/) projects, add the astro extension too:
...
 "editor.codeActionsOnSave": {
     "source.organizeImports": false
 }
 ```
+
+If you wish, you can automatically fix some ESLint violations on save, including sorting imports:
+
+```JSONC title=".vscode/settings.json"
+"editor.codeActionsOnSave": {
+    "source.fixAll.eslint": "explicit"
```
Comment (AndreaPontrandolfo) [inline apps/docs-website/docs/vscode-support.md:53]: So, i know that a lot of other configs recommend this in the docs, but up until now i purposefully didn't include it.
The reason being, is that this is a generic ESLint usage recommendation which has nothing to do specifically with Sheriff, so i didn't want to clutter the docs with irrelevant informations, even though it could be useful for newbies approaching ESLint for the first time.
What do you think of my reasoning here?

File: apps/docs-website/docs/vscode-support.md [hunk 4]
```diff
@@ -38,10 +38,18 @@ For [Astro](https://astro.build/) projects, add the astro extension too:
 
 ## Avoid "source.organizeImports"
 
-Sheriff already handle imports sorting, so you have to disable thew VSCode automatic import sorting feature to avoid conflicts:
+Sheriff already handle imports sorting, so you have to disable VSCode automatic import sorting feature to avoid conflicts:
 
 ```JSONC title=".vscode/settings.json"
 "editor.codeActionsOnSave": {
     "source.organizeImports": false
```
Comment (AndreaPontrandolfo) [inline apps/docs-website/docs/vscode-support.md:45]: actually false will be deprecated. We shall recommend `never`

File: apps/docs-website/docs/vscode-support.md [hunk 5]
```diff
@@ -38,10 +38,18 @@ For [Astro](https://astro.build/) projects, add the astro extension too:
 
 ## Avoid "source.organizeImports"
 
-Sheriff already handle imports sorting, so you have to disable thew VSCode automatic import sorting feature to avoid conflicts:
+Sheriff already handle imports sorting, so you have to disable VSCode automatic import sorting feature to avoid conflicts:
```
Comment (AndreaPontrandolfo) [inline apps/docs-website/docs/vscode-support.md:41]: I think this is slightly misleading. We should probably reword this as: "Sheriff already handle imports sorting, so if you happen to have enabled the VSCode automatic import sorting feature, you should disable it to avoid conflicts"

**Review by AndreaPontrandolfo — CHANGES_REQUESTED** [review]

File: packages/eslint-config-sheriff/src/getReactConfig.ts (hunk 1 above)
Comment (lishaduck) [reply packages/eslint-config-sheriff/src/getReactConfig.ts:60]: A11y should be strict by default. I looked and they seem to be the same right now, but it would potentially add stricter linting in the far-flung future.

File: packages/eslint-config-sheriff/src/getBaseConfig.ts (hunk 2 above)
Comment (lishaduck) [reply packages/eslint-config-sheriff/src/getBaseConfig.ts:139]: The linked issue said to add it, so I did. 🤷‍♂️

File: apps/docs-website/docs/vscode-support.md (hunk 4 above)
Comment (lishaduck) [reply apps/docs-website/docs/vscode-support.md:45]: Oh, yeah, I should've caught that.

File: apps/docs-website/docs/vscode-support.md (hunk 3 above)
Comment (lishaduck) [reply apps/docs-website/docs/vscode-support.md:53]: Yeah, but with import sorting I think it makes sense, as otherwise it's not a direct replacement to codeActionsOnSave.

File: apps/docs-website/docs/vscode-support.md (hunk 5 above)
Comment (lishaduck) [reply apps/docs-website/docs/vscode-support.md:41]: nit: "handles", not "handle"

Otherwise, that makes sense.

File: apps/docs-website/docs/vscode-support.md (hunk 3 above)
Comment (AndreaPontrandolfo) [reply apps/docs-website/docs/vscode-support.md:53]: Agree

File: packages/create-sheriff-config/src/utils/setSheriffConfig.ts [hunk 6]
```diff
@@ -1,12 +1,13 @@
 import { highlight } from 'cli-highlight';
 import { consola } from 'consola';
 import { sheriffStartingOptions } from '@sherifforg/constants';
+import type { SheriffConfigurablePlugins } from '../../../sheriff-types/src';
```
Comment (AndreaPontrandolfo) [inline packages/create-sheriff-config/src/utils/setSheriffConfig.ts:4]: the correct import is `@sherifforg/types`

File: packages/eslint-config-sheriff/src/getAstroConfig.ts [hunk 7]
```diff
@@ -1,30 +1,40 @@
...
-    plugins: {
-      astro,
-    },
-    languageOptions: {
-      parser: astroParser,
-      parserOptions: {
-        parser: '@typescript-eslint/parser',
-        project: customTSConfigPath || true,
-        extraFileExtensions: ['.astro'],
+): FlatConfig.ConfigArray => {
```
Comment (AndreaPontrandolfo) [inline packages/eslint-config-sheriff/src/getAstroConfig.ts:8]: i think we should remove this. We are okay with the inferred types, plus i already tried to use `@typescript-eslint` types in the past and it didn't end well, because they don't get exported along the rest of the package and end up erroring out in the .dts build.

File: packages/eslint-config-sheriff/src/getBaseConfig.ts (hunk 2 above)
Comment (AndreaPontrandolfo) [reply packages/eslint-config-sheriff/src/getBaseConfig.ts:139]: I think this won't hurt, even though w don't really support .js files.

**Review by AndreaPontrandolfo — CHANGES_REQUESTED** [review]

File: packages/eslint-config-sheriff/src/getReactConfig.ts (hunk 1 above)
Comment (AndreaPontrandolfo) [reply packages/eslint-config-sheriff/src/getReactConfig.ts:60]: you are right, they seem to overlap 100%.

File: packages/create-sheriff-config/src/utils/setSheriffConfig.ts (hunk 6 above)
Comment (lishaduck) [reply packages/create-sheriff-config/src/utils/setSheriffConfig.ts:4]: 🤦‍♂️

File: packages/eslint-config-sheriff/src/getAstroConfig.ts (hunk 7 above)
Comment (lishaduck) [reply packages/eslint-config-sheriff/src/getAstroConfig.ts:8]: > i think we should remove this. We are okay with the inferred types, plus i already tried to use `@typescript-eslint` types in the past and it didn't end well, because they don't get exported along the rest of the package and end up erroring out in the .dts build.

Huh. I was running into the opposite, the .d.ts build would fail to build _without_ explicitly setting it becuase it wasn't inferred correctly, but for some reason it worked with it.

Comment (AndreaPontrandolfo) [reply packages/eslint-config-sheriff/src/getAstroConfig.ts:8]: Interesting. I think I'll look at this closely myself then.

Comment (lishaduck) [reply packages/eslint-config-sheriff/src/getAstroConfig.ts:8]: Oh, to be clear, I don't think it was on this branch, but I recall there being an issue with functions returning configs getting incorrect types inferred.

Comment (lishaduck) [reply packages/eslint-config-sheriff/src/getAstroConfig.ts]: I removed it. Seemed to work fine for now.

**Review by AndreaPontrandolfo — APPROVED** [review]
